.github/workflows/release.yml

```diff
@@ -0,0 +1,111 @@
+        with:
+          test-bot: false
+
+      - name: Determine tag
+        id: determine-tag
+        run: |
+          brew ruby -e "$(cat<<EOS
+          require "formula"
+
+          tag = Formula["portable-ruby"].pkg_version.to_s
+
+          existing_tag = Utils.safe_popen_read("git", "tag", "--list", tag)
+          raise "Tag #{tag} already exists" if existing_tag.present?
+
+          puts "Preparing for #{tag} release."
+          puts "::set-output name=tag::#{tag}"
+          EOS
+          )"
+
+  build:
+    needs: determine-tag
+    uses: ./.github/workflows/build.yml
+
+  release:
+    needs: [determine-tag, build]
+    runs-on: ubuntu-latest
+    env:
+      TAG: ${{ needs.determine-tag.outputs.tag }}
+    steps:
+      - name: Set up Homebrew
+        uses: Homebrew/actions/setup-homebrew@master
+        with:
+          test-bot: false
+
+      - name: Checkout branch
+        run: git checkout "${{ github.ref_name }}"
+
+      - name: Install gems
+        run: brew install-bundler-gems
+
+      - name: Configure Git user
+        uses: Homebrew/actions/git-user-config@master
+
+      - name: Set up commit signing
+        uses: Homebrew/actions/setup-commit-signing@master
+        with:
+          signing_key: ${{ secrets.BREWTESTBOT_GPG_SIGNING_SUBKEY }}
+
+      - name: Setup directory
+        run: mkdir bottles
+
+      - name: Download bottles
+        uses: actions/download-artifact@v3
+        with:
+          name: bottles
+          path: bottles
+
+      - name: Upload to GitHub Packages
+        working-directory: bottles
+        env:
+          BREWTESTBOT_NAME_EMAIL: "BrewTestBot <1589480+BrewTestBot@users.noreply.github.com>"
+          HOMEBREW_GPG_PASSPHRASE: ${{ secrets.BREWTESTBOT_GPG_SIGNING_SUBKEY_PASSPHRASE }}
+          HOMEBREW_GITHUB_PACKAGES_USER: ${{ github.actor }}
+          HOMEBREW_GITHUB_PACKAGES_TOKEN: ${{ github.token }}
+        run: brew pr-upload --debug --committer="${BREWTESTBOT_NAME_EMAIL}" --root-url=https://ghcr.io/v2/${GITHUB_REPOSITORY_OWNER,,}/portable-ruby
+
+      - name: Push commits
+        uses: Homebrew/actions/git-try-push@master
+        with:
+          token: ${{ github.token }}
+          branch: ${{ github.ref_name }}
+        env:
+          GIT_COMMITTER_NAME: BrewTestBot
+          GIT_COMMITTER_EMAIL: 1589480+BrewTestBot@users.noreply.github.com
+          HOMEBREW_GPG_PASSPHRASE: ${{ secrets.BREWTESTBOT_GPG_SIGNING_SUBKEY_PASSPHRASE }}
+
+      - name: Push tag
+        env:
+          GITHUB_TOKEN: ${{ github.token }}
+        run: |
+          gh release create "${TAG}" --notes "Portable Ruby ${TAG}" --target "$(git rev-parse --verify HEAD)"
+
+      - name: Upload to GitHub Releases
+        working-directory: bottles
+        env:
+          HOMEBREW_GITHUB_API_TOKEN: ${{ github.token }}
+        run: brew pr-upload --debug --upload-only --root-url=https://github.com/${GITHUB_REPOSITORY}/releases/download/${TAG}
```

Comment:
Completely overkill but: might be nice if this could create the Homebrew/brew PR or output some instructions on what needs to be entered and in what file?

Reply:
> Some sort of brew automation would be nice, but I'd be more comfortable to first change things around on that end to be easier to machine-replace (and ideally in a format that'll never really change for style reasons etc).

👍🏻 agreed.